lib/testers/rule-tester.js

```diff
@@ -490,7 +490,19 @@ RuleTester.prototype = {
                 }
 
                 if (item.hasOwnProperty("output")) {
-                    const fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+                    let fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+
+                    if (item.multipass) {
+                        for (let i = 0; i < 10; i++) {
+                            const newItem = Object.assign({}, item, { code: fixResult.output });
+                            const newResult = runRuleForItem(ruleName, newItem);
+
+                            fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), newResult.messages);
+                            if (newItem.code === fixResult.output) {
+                                break;
```

Comment:
If we don't break here, we'd be running the rule again on exactly the same code as in the previous iteration, so it should again get no fixes assuming the rules are deterministic. So I feel like there isn't really a use case for continuing, but I might be overlooking something.

(Another thought I had was to fail the test after a certain number of iterations, since I feel like it indicates a bug in a rule if it doesn't reach a stable point after some number of fixes.)